src/cache-restore.ts

```diff
@@ -29,7 +29,8 @@ export const restoreCache = async (
     );
   }
 
-  const primaryKey = `setup-go-${platform}-go-${versionSpec}-${fileHash}`;
+  const imageOS = process.env.ImageOS || 'self-hosted';
+  const primaryKey = `setup-go-${platform}-${imageOS}-go-${versionSpec}-${fileHash}`;
```

Comment:
I agree with you. I think the solution should be limited only of Linux. Could you please change it.